sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CommitDiscardWrapperComponent.java

```diff
@@ -437,7 +437,9 @@ public Button getDeleteWithReasonOrUndeleteButton(String entityName, boolean del
 						String.format(I18nProperties.getString(Strings.confirmationDeleteEntity), entityName),
 						this::onDeleteWithReason);
 				} else {
-					onDeleteWithReason(null);
+					DeletableUtils.showDeleteWithReasonPopup(
```

Comment:
can you please explain this
indeed i have included the undelete action within the delete listeners as i didn't want to change everything regarding the delete process
but do we need a popup on undelete ?

Reply:
No, you're right. I've got confused with the deleted and !deleted flags since (deleted) reffers to the fact that undeleted action is necessary not the fact that delete will be used. 

I'll do the change. :)